lib/timers.js

```diff
@@ -477,47 +477,109 @@ exports.clearImmediate = function(immediate) {
 
 var unrefList, unrefTimer;
 
+function _makeTimerTimeout(timer) {
+  var domain = timer.domain;
+  var msecs = timer._idleTimeout;
+
+  // Timer has been unenrolled by another timer that fired at the same time,
+  // so don't make it timeout.
+  if (!msecs || msecs < 0)
```

Comment:
`timer._idleTimeout` is always a number, isn't it?  I think the condition can be simplified here to just `msecs <= 0`.